app/controllers/store_controller.rb

```diff
@@ -149,7 +154,11 @@ def process_donation
     redirect_route = quick_donate_path(:customer_id => @customer.id, :account_code_string => @account_code.code, :donation => @amount)
     @amount > 0 or return redirect_to(redirect_route, :alert => 'Donation amount must be provided')
     # Given valid donation, customer, and charge token, create & place credit card order.
-    @gOrderInProgress = Order.new_from_donation(@amount, @account_code, @customer)
+    @gOrderInProgress = Order.new_from_donation(@amount, Donation.default_code, @customer)
+    case params[:donation_frequency]
+    when DONATION_FREQUENCIES[1]
```

Comment:
this isn't legible.  `DONATION_FREQUENCIES[1]` isn't meaningful unless you go look up how the array is defined, which defeats the purpose of good naming. i would just define two constants, `DONATION_ONE_TIME` and `DONATION_RECURRING`, as the two possible valid values.

Reply:
changed